### DISCUSSION TITLE
Enhancement for IEnumerable<Option<TIn>>

### DISCUSSION THREAD

**Discussion by LaszloLueck** [louthy/language-ext#1033] [General]
Hi,
very nice Toolset and keep up your fantastic work!
I switch between Scala and C# all the time and you build exactly the bridge I need!
I have a question / extension or maybe I haven't found the right function yet that does what I'm looking for:
Sometimes I have a list of elements e.g.
`var list = List(Some(1), None, Some(2), None, Some(3));`

Of course, I want to transform this list at some point, so that only the list of elements remains.
So a `List<Option<TIn>>` resulted in `List<TIn>`

I have now written a function extension for this:
```
    private static IEnumerable<TIn> Values<TIn>(this IEnumerable<Option<TIn>> source)
    {
        return source
            .Filter(f => f.IsSome)
            .Select((a, _) => a.ValueUnsafe());
    }
```
And for a quick usage example:
```
var list = List(Some(1), None, Some(2), None, Some(3));

foreach (var i in list.Values())
{
     Console.WriteLine(i);
}

returns:
1
2
3
```

Is there anything in LanguageExt for this? If not, I would push this into the project as a merge request?

Greetings!

**Comment by LaszloLueck** [comment]
Aha,
found the internal solution
`List.Somes()`

does exactly what i need.

Please close this one!

Regards,
Laszlo

**Comment by louthy** [comment]
That's the one - also `xs.Lefts()`, `xs.Rights()`, and `xs.Partition()` for `IEnumerable<Either<L, R>>`
